cmd/utils/packs.go

```diff
@@ -23,8 +23,9 @@ var nameRegex = regexp.MustCompile(fmt.Sprintf("^%s$", namePattern))
 
 // versionPattern validates pack version.
 // Ref: https://github.com/ARM-software/CMSIS_5/blob/develop/CMSIS/Utilities/PackIndex.xsd
-//                    <major>           . <minor>          . <patch>            - <quality>                                                                                             + <meta info>
-var versionPattern = `(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*)?(?:\+[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*)?`
+// With little adjustments to reduce the number of capturing groups to a single one
+//                       <major>     . <minor>       . <patch>         - <quality>                                                                                       + <meta info>
+var versionPattern = `(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(?:(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?:[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?`
```

Comment:
Where did you get this pattern from? Maybe worth putting it in the comment for later reference.

Reply:
Oh, it's confusing because I mentioned in the Issue that I swapped the regex with the one from SemVer.org. But at the end I kept using the original regex from PackIndex.xsd, there is a URL reference on line 25 above.

I changed the original regex a little bit in order to fix the bug.

Reply:
Okay, but I still think we need to get the fix into the PackIndex.xsd and potentially in PACK.xsd as well.
 Or am I missing something.

Reply:
Yes, I agree with that. The regex works just fine if you use it just to validate a given version string. The problem starts only when you use it to extract the bits out. I'll open a PR with a fixed for PACK.xsd